File: src/ML2Typescript/Examples/testWith.ml
type point = { x: int; y: int }

let pt = { x = 3; y = 5 };;

let init (z: int): point = { x = z; y = z };;

let pt2 = { pt with y = 9 };;

let pt3 = { (init 10) with x = 10 };;

let _ = print_int pt.y;;

let _ = print_int pt2.y;;

let _ = print_int pt3.y;;

File: src/ML2Typescript/Examples/TS/testWith.ts
import { Option, none, some } from './Lib.js';
import * as ArrayList from './ArrayList.js';

type Point = {x: number, y: number}
const pt = { x: 3, y: 5 };

function init(z: number): Point {
    return { x: z, y: z };
}
const pt2 = { ...pt, y: 9 };
const pt3 = { ...init(10), x: 10 };
console.log(pt.y);
console.log(pt2.y);
console.log(pt3.y);
